src/spu-schema/src/server/stream_fetch.rs

```diff
@@ -86,10 +95,53 @@ impl Default for SmartStreamKind {
 #[derive(Debug, Clone, Encoder, Decoder)]
 pub enum SmartStreamWasm {
     Raw(Vec<u8>),
+    /// compressed WASM module payload using Gzip
+    #[fluvio(min_version = 14)]
+    Gzip(Vec<u8>),
     // TODO implement named WASM modules once we have a WASM store
     // Url(String),
 }
 
+fn zip(raw: Vec<u8>) -> io::Result<Vec<u8>> {
+    let mut encoder = GzEncoder::new(raw.as_ref(), Compression::default());
+    let mut buffer = Vec::with_capacity(raw.len());
+    encoder.read_to_end(&mut buffer)?;
+    Ok(buffer)
+}
+
+fn unzip(compressed: Vec<u8>) -> io::Result<Vec<u8>> {
+    let mut decoder = GzDecoder::new(compressed.as_ref());
+    let mut buffer = Vec::with_capacity(compressed.len());
+    decoder.read_to_end(&mut buffer)?;
+    Ok(buffer)
+}
+
+impl SmartStreamWasm {
+    /// returns the compressed WASM module bytes
+    pub fn to_compressed(self) -> io::Result<Self> {
```

Comment:
I might call this `to_gzip` in case we decide to switch compression schemes later